Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -0,0 +1,494 @@
+     */
+    private $factoryShippingOptionsMock;
+
+    /**
+     * @var ShippingTaxInterfaceFactory
+     */
+    private $shippingTaxInterfaceFactory;
+
+    /**
+     * @var ShippingOptionInterfaceFactory
+     */
+    private $shippingOptionInterfaceFactory;
+
+    /**
+     * @var TotalsCollector
+     */
+    private $totalsCollector;
+
+    /**
+     * @var ShippingMethodConverter
+     */
+    private $converter;
+
+    /**
+     * @var logHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var BoltErrorResponse
+     */
+    private $errorResponse;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var PriceHelper
+     */
+    private $priceHelper;
+
+    /**
+     * @var SessionHelper
+     */
+    private $sessionHelper;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->factoryShippingOptionsMock = $this->getMockBuilder(ShippingOptionsInterfaceFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->shippingTaxInterfaceFactory = $this->getMockBuilder(ShippingTaxInterfaceFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->shippingOptionInterfaceFactory = $this->getMockBuilder(ShippingOptionInterfaceFactory::class)
+            ->setMethods(['create'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->hookHelper = $this->createMock(HookHelper::class);
+        $this->regionModel = $this->createMock(RegionModel::class);
+        $this->totalsCollector = $this->createMock(TotalsCollector::class);
+        $this->converter = $this->createMock(ShippingMethodConverter::class);
+        $this->logHelper = $this->createMock(LogHelper::class);
+        $this->errorResponse = $this->createMock(BoltErrorResponse::class);
+        $this->response = $this->createMock(Response::class);
+        $this->configHelper = $this->createMock(ConfigHelper::class);
+        $this->request = $this->createMock(Request::class);
+        $this->cache = $this->createMock(CacheInterface::class);
+        $this->priceHelper = $this->createMock(PriceHelper::class);
+        $this->sessionHelper = $this->createMock(SessionHelper::class);
+
+        $this->bugsnag = $this->getMockBuilder(Bugsnag::class)
+            ->setMethods(['notifyException'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->bugsnag->method('notifyException')
+            ->willReturnSelf();
+    }
+
+    public function testGetShippingMethodsIfQuoteEmptyReturnException()
```

Comment:
This is a style choice, but I'm a big fan of using PHPUnit's annotations. You can tag methods with `@test` so you don't need to prefix each test function, and here you can use `@expectedException` and `@expectedExceptionMessage` for assertions, making the purpose of this test really clear without having to scan the body.

Reply:
gotcha, the code was refactored.